kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/file/matchers.kt

```diff
@@ -233,3 +233,24 @@ fun containFileDeep(name: String): Matcher<Path> = object : Matcher<Path> {
       fileExists(value.toFile()), "File $name should exist in $value", "File $name should not exist in $value"
   )
 }
+
+fun Path.shouldContainFiles(vararg files: String) = this should containsAllFiles(files.asList())
+fun Path.shouldNotContainFiles(vararg files: String) = this shouldNot containsAllFiles(files.asList())
+fun containsAllFiles(files: List<String>) = object : Matcher<Path> {
```

Comment:
Can we call this containFiles so then people can do `dir should containFiles(a, b)` which reads a bit nicer.